fibsem/milling/patterning/plotting.py

```diff
@@ -530,11 +697,97 @@ def draw_rectangle_shape(pattern_settings: FibsemRectangleSettings, image_shape:
 
     return DrawnPattern(pattern=shape, position=pos, is_exclusion=pattern_settings.is_exclusion)
 
+
+def draw_line_shape(
+    pattern_settings: FibsemLineSettings,
+    image_shape: Tuple[int, int],
+    pixelsize: float,
+) -> DrawnPattern:
+    from skimage.draw import line_aa
+
+    start_pixel_x, start_pixel_y, end_pixel_x, end_pixel_y = (
+        _line_pattern_to_image_pixels(
+            pattern_settings, pixelsize, (image_shape[0], image_shape[1])
+        )
+    )
+
+    array_max = (
+        int(round(end_pixel_y - start_pixel_y)),
+        int(round(end_pixel_x - start_pixel_x)),
+    )
+
+    shape = np.zeros((array_max[0] + 1, array_max[1] + 1), dtype=float)
+
+    rr, cc, val = line_aa(0, 0, array_max[0], array_max[1])
+    shape[rr, cc] = val
+
+    cx = int(round((start_pixel_x + end_pixel_x) / 2))
+    cy = int(round((start_pixel_y + end_pixel_y) / 2))
+
+    pos = Point(x=cx, y=cy)
+
+    return DrawnPattern(pattern=shape, position=pos, is_exclusion=False)
+
+
+def draw_bitmap_shape(
+    pattern_settings: FibsemBitmapSettings,
+    image_shape: Tuple[int, int],
+    pixelsize: float,
+) -> DrawnPattern:
+    from PIL import Image
+
+    # image parameters (centre, pixel size)
+    icy, icx = image_shape[0] // 2, image_shape[1] // 2
+    pixelsize_x, pixelsize_y = pixelsize, pixelsize
+
+    # pattern parameters
+    width = pattern_settings.width
+    height = pattern_settings.height
+    centre_x = pattern_settings.centre_x
+    centre_y = pattern_settings.centre_y
+    rotation = pattern_settings.rotation
+
+    # pattern to pixel coords
+    w = int(width / pixelsize_x)
+    h = int(height / pixelsize_y)
+    cx = int(icx + (centre_x / pixelsize_x))  # Fix: use pixelsize_x for x coordinate
+    cy = int(icy - (centre_y / pixelsize_y))
+
+    bitmap = pattern_settings.bitmap
+    
+    if bitmap is None:
+        bitmap = np.zeros((1, 1, 2), dtype=float)
+
+    image_bmp = Image.fromarray(bitmap[:, :, 0].squeeze().astype(float))
+
+    if pattern_settings.flip_y:
+        image_bmp = image_bmp.transpose(Image.Transpose.FLIP_TOP_BOTTOM)
+
+    image_resized = image_bmp.resize((w, h))
+
+    if not np.isclose(rotation, 0):
+        image_rotated = image_resized.rotate(-pattern_settings.rotation, expand=True)
+
+    # Create base rectangle shape
+    shape = np.asarray(image_rotated, dtype=np.float_)
```

Comment:
Doesn't work when rotation = 0
```suggestion
    if not np.isclose(rotation, 0):
        image_resized = image_resized.rotate(-pattern_settings.rotation, expand=True)

    # Create base rectangle shape
    shape = np.asarray(image_resized, dtype=np.float_)
```